lightning/src/routing/router.rs

```diff
@@ -633,6 +649,10 @@ pub fn get_route<L: Deref>(our_node_pubkey: &PublicKey, network: &NetworkGraph,
 					let over_path_minimum_msat = amount_to_transfer_over_msat >= $directional_info.htlc_minimum_msat &&
 						amount_to_transfer_over_msat >= $next_hops_path_htlc_minimum_msat;
 
+					let next_hops_penalty_msat = $next_hops_penalty_msat
```

Comment:
We can move this at least down below the next if to avoid calling the penalty lookup unless we need to (and I believe down another if block as well, see below).